### ISSUE TITLE
[REQUEST] Relax pinned library versions

### ISSUE THREAD

**Issue by snazzyfox** [interactions-py/interactions.py#441]
**What is the feature you're proposing?**

## Summary

TL;DR: Relax version pins for dependencies of this library. 

For libraries, pinning specific versions severely limits users' options. Common industry practice is to pin to the required major versions, and exclude minor ranges if necessary.

An example of this is [aiohttp's version specs](https://github.com/aio-libs/aiohttp/blob/master/setup.cfg#L49-L57) itself.

#### Current state

Pinning a specific version requires users and all other installed libraries in the same project to use that specific version. It eliminates the possibility to use a different but compatible version, and removes the possibility to use any libraries that are incompatible with the pinned version, even if another compatible versions exist. For example:
```
discord-interactions
|-- aiohttp==3.7.4post0
some-other-lib
|-- aiohttp>=3.8.0,<4.0.0
```
It takes significant efforts for users to bypass version constraints since they need to work *against* any version resolution logic provided in packaging tools. This may even be impossible for certain tools or during CI due to limited ability to monkeypatch the process. Every upgrade also becomes risky for users because it can introduction new unnecessary incompatibilities.

#### Desired state

The dependency should be defined as:
```
discord-interactions
|-- aiohttp>=3.7.4,<4.0.0
some-other-lib
|-- aiohttp>=3.8.0,<4.0.0
```

This allows users to freely select any compatible version between the declared ranges in case a different but compatible version is needed, while at the same time preventing breaking changes (signified by a major version bump).

I understand that there is a risk here since there is no unit/integration testing to catch these issues. However, it is still the better option, because it is trivial for users to temporarily backpin the breaking dependency, and also trivial for us to mark versions as incompatible as they come up.

I'm happy to submit a PR for this if you all are open to the idea.

**Additional Information**
- [ ] My feature request is related to an existing Issue.
    - Issue (if referenceable):

**Comment by i0bs** [comment]
I believe @DeltaXWizard brought this to my attention a month ago when we were making some progress on 4.0.1. This has been noted and we will be making reflecting changes to the `requirements.txt` moving forward to no longer hard-lock module versions.
